src/App.js

```diff
@@ -1,16 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const chatMessagesInitialCopy = chatMessages.map((chat) => {
+    return { ...chat };
+  });
+
+  const [chatsList, setChatsList] = useState(chatMessagesInitialCopy);
+
+  const updateLike = (id) => {
+    // console.log(`updateLike called on chat id: ${id}`);
+    const newArr = [];
+    for (const chat of chatsList) {
+      if (id !== chat.id) {
+        newArr.push(chat);
+      } else {
+        const newChat = {
+          ...chat,
+          liked: !chat.liked,
+        };
+        newArr.push(newChat);
+      }
+    }
+    setChatsList(newArr);
+  };
+
+  const countLikes = () => {
+    let count = 0;
+    for (const chat of chatsList) {
+      if (chat.liked === true) {
+        count += 1;
+      }
+    }
+    return count;
+  };
```

Comment:
Storing the number of liked messages in a state variable would make it possible to make this an O(1) operation rather than an O(n). Every time updateLike is called, one is effectively added or subtracted from the number of likes.